### PR TITLE
Move "Enable Opensea API" and "NFT Autodetect" into Security & Privacy settings

### LINKED ISSUE #20062: Move Enable Opensea API and audotedect NFTs into Security & Privacy settings
## Background
Enable Opensea API and autodetect NFTs from Experimental to Security & Privacy settings. While Enable Opensea API and autodetect NFTs was under the Experimental settings, we've made minor improvements and the amount of reported bugs have been minimal. Therefore, we're going to move this setting out of experimental and into Security and Privacy.

## Acceptance Criteria
Scenario: As a user on the Settings screen
- Given that the user is on the Security & Privacy settings screen
- When the user scrolls down to Enable Opensea API and autodetect NFTs toggles
- Then retain the same functionality to enable Opensea API and autodetecting of NFTs
- Update e2e tests and jest snapshots
- Update the settings ref for settings search

## Designs
For listing ordering of privacy settings, go [here](https://www.figma.com/file/Ek7QDiB3TgvDhjPtmh1zAT/Privacy-Epic-Security%26Privacy-settings?type=design&node-id=368-52616&mode=design&t=Cqhn5gwX1TPDYNL9-0).

### PR DESCRIPTION
**Pull request by vthomas13**
## Explanation
Moves existing Opensea API and NFT Autodetect settings from the Experimental tab to the Security & Privacy tab
Fixes https://github.com/MetaMask/metamask-extension/issues/20062
Progress: https://github.com/MetaMask/metamask-extension/issues/20062#issuecomment-1654196908

### Before
image
image

### After

image
image

## Manual Testing Steps

Open extension; go to Settings
Verify that the settings ARE NOT in Experimental tab
Verify that the settings ARE in Security & Privacy tab
Verify that searching with settings search bar will link directly to the setting

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]
Looks good on the Settings Page. I think we need to fix it on home page



https://github.com/MetaMask/metamask-extension/assets/39872794/72c8bbc9-790e-4325-ad33-8332cf215069

**Comment by vthomas13** [comment]
Thanks @NidhiKJha, great catch!

**Review by pedronfigueiredo — DISMISSED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
+  "[javascript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
```
Comment (darkwing) [inline .vscode/settings.json:3]: I don't think this was meant to hit the PR

File: app/scripts/controllers/preferences.js [hunk 2]
```diff
@@ -178,17 +189,6 @@ export default class PreferencesController {
...
-   * Setter for the `openSeaEnabled` property
-   *
-   * @param {boolean} openSeaEnabled - Whether or not the user prefers to use the OpenSea API for NFTs data.
-   */
-  setOpenSeaEnabled(openSeaEnabled) {
-    this.store.updateState({
-      openSeaEnabled,
-    });
-  }
-
```
Comment (darkwing) [inline app/scripts/controllers/preferences.js:191]: Is this change required?  It looks exactly the same -- if it's the same, I don't think we need the change, as it makes the PR larger than necessary.

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 3]
```diff
@@ -239,44 +142,44 @@ export default class ExperimentalTab extends PureComponent {
...
     return (
       <>
-        <Typography
-          variant={TypographyVariant.H4}
+        <Text
+          variant={TextVariant.headingSm}
           color={TextColor.textAlternative}
           marginBottom={2}
-          fontWeight={FONT_WEIGHT.BOLD}
+          fontWeight={FontWeight.Bold}
```
Comment (darkwing) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:149]: Thank you so much for replacing these deprecated values!

**Review by darkwing — CHANGES_REQUESTED** [review]

File: app/scripts/controllers/preferences.js (hunk 2 above)
Comment (vthomas13) [reply app/scripts/controllers/preferences.js:191]: Oh yeah, this change is unnecessary now that I removed the clearKnownMethodData method

File: .vscode/settings.json (hunk 1 above)
Comment (vthomas13) [reply .vscode/settings.json]: fixed in 7ab324e3a57411239fdfec94d09b11f388c3c97f

**Comment by darkwing** [comment]
For some reason this link keeps going to Experimental
SCR-20230802-kmtb

**Comment by vthomas13** [comment]
@darkwing fixed in 474a74980d554f8056bafbcdc1a7bec2244dfc80

> For some reason this link keeps going to Experimental SCR-20230802-kmtb

**Review by NidhiKJha — APPROVED** [review]
LGTM

**Review by Ameralameri — APPROVED** [review]
Approv

**Review by NidhiKJha — DISMISSED** [review]
The routing seems to be broken here. When I click on Turn NFT Detection on, it takes me to Security page but not to the toggle



https://github.com/MetaMask/metamask-extension/assets/39872794/e832c9c6-ed4d-4b4c-af52-e361fe3350bb

**Comment by vthomas13** [comment]
> The routing seems to be broken here. When I click on Turn NFT Detection on, it takes me to Security page but not to the toggle
> 
>  257893812-ee0b1cbd-4fc0-4bef-b37d-531174ba6c17.mov

Nice catch, I think it's due to a settingsRef collision with another branch. We may want to look into refactoring that in the future. Numerical keys may not be the best approach here, but I'm not sure. What do you think @darkwing?

**Review by darkwing — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]
